src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -0,0 +1,121 @@
+import { OnChange, Values } from "../PropertyDecorators";
+import { IsDisplayable } from "../ClassDecorators";
+import { SelfServeBaseClass } from "../SelfServeUtils";
+import { ChoiceItem, InputType, NumberUiType, SmartUiInput } from "../../Explorer/Controls/SmartUi/SmartUiComponent";
+import {
+  getRegionSpecificSku,
+  initializeDedicatedGatewayProvisioning,
+  Sku,
+  updateDedicatedGatewayProvisioning,
+  refreshDedicatedGatewayProvisioning,
+} from "./SqlX.rp";
+import { RefreshResult, SelfServeNotification } from "../SelfServeComponent";
+import { MessageBarType } from "office-ui-fabric-react";
+
+const onEnableDedicatedGatewayChange = (
+  currentState: Map<string, SmartUiInput>,
+  newValue: InputType
+): Map<string, SmartUiInput> => {
+  const sku = currentState.get("sku");
+  const instances = currentState.get("instances");
+  const isSkuHidden = newValue === undefined || !(newValue as boolean);
+  currentState.set("enableDedicatedGateway", { value: newValue });
+  currentState.set("sku", { value: sku.value, hidden: isSkuHidden });
+  currentState.set("instances", { value: instances.value, hidden: isSkuHidden });
+  return currentState;
+};
+
+const getSkuLabel = (sku: Sku): string => {
+  switch (sku) {
+    case Sku.D4:
+      return "Cosmos D4s";
+    case Sku.D8:
+      return "Cosmos D8s";
+    case Sku.D16:
+      return "Cosmos D16s";
+    case Sku.D32:
+      return "Cosmos D32s";
+    case Sku.D64:
+      return "Cosmos D64s";
+    default:
+      return "Unsupported Sku";
+  }
+};
+
+const getSkus = async (): Promise<ChoiceItem[]> => {
+  const skus = await getRegionSpecificSku();
+  return skus.map((sku) => {
+    return { label: getSkuLabel(sku), key: sku };
+  });
+};
+
+@IsDisplayable()
+export default class SqlX extends SelfServeBaseClass {
+  public onRefresh = async (): Promise<RefreshResult> => {
+    return refreshDedicatedGatewayProvisioning();
+  };
+
+  public validate = (currentvalues: Map<string, SmartUiInput>): string => {
+    if (!currentvalues.get("sku").value || !currentvalues.get("instances").value) {
+      return "SKU and instances should not be empty.";
+    }
+    return undefined;
+  };
+
+  public onSave = async (currentValues: Map<string, SmartUiInput>): Promise<SelfServeNotification> => {
+    const enableDedicatedGateway = currentValues.get("enableDedicatedGateway")?.value as boolean;
+    if (enableDedicatedGateway) {
+      const sku = Sku[currentValues.get("sku")?.value as keyof typeof Sku];
+      const instances = currentValues.get("instances")?.value as number;
+      await updateDedicatedGatewayProvisioning(sku, instances);
+    } else {
+      await updateDedicatedGatewayProvisioning(undefined, undefined);
+    }
+    return { message: "submitted sqlX update successfully", type: MessageBarType.info };
+  };
+
+  public initialize = async (): Promise<Map<string, SmartUiInput>> => {
```

Comment:
>initialize [](start = 9, length = 10)

nit: put it as first method within the class.


initialize, onSave, onRefresh